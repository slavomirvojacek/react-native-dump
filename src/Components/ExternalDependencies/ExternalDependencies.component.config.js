import PropTypes from 'prop-types'

const displayName = 'ExternalDependencies'

const propTypes = {
  logIntoConsole: PropTypes.func.isRequired
}

export default {
  displayName, propTypes
}
