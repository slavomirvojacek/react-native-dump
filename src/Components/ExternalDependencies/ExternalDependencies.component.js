import { throwIfNotFunction } from '@openmaths/utils'
import React, { Component } from 'react'
import { Text } from 'react-native'

import SomethingExternal from './lib/SomethingExternal'
import SomethingElseExternal from './lib/SomethingElseExternal'

class ExternalDependencies extends Component {
  static displayName = 'ExternalDependencies'

  componentDidMount () {
    const { logIntoConsole } = this.props

    throwIfNotFunction(logIntoConsole, '`logIntoConsole` has to be a Function', TypeError)

    logIntoConsole('value')
  }

  someHandler = value => {
    SomethingExternal.awesomeMethod(value)
    SomethingElseExternal.anotherAwesomeMethod(value)
  }

  render () {
    return (
      <Text>Some Text</Text>
    )
  }
}

export default ExternalDependencies
