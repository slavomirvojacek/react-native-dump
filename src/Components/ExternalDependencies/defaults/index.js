import { noop } from '@openmaths/utils'

const props = {
  logIntoConsole: noop,
}

export {
  props
}
