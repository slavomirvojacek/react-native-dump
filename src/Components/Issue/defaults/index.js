const issueDefault = (override = {}) => {
  const def = {
    id: '',
    title: '',
    url: '',
    bodyText: '',
    reactions: []
  }

  return {...def, ...override}
}

const props = {
  issue: issueDefault(),
}

export {
  issueDefault, props
}
