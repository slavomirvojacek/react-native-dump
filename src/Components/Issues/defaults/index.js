const props = {
  issues: [],
}

export {
  props
}
