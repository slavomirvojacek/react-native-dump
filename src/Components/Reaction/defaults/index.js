const userDefault = (id = '') => ({id})

const reactionDefault = (override = {}) => {
  const def = {
    id: '',
    content: '',
    user: userDefault()
  }

  return {...def, ...override}
}

const props = {
}

export {
  reactionDefault, props
}
