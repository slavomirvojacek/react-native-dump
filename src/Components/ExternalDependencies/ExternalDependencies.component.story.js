import React from 'react'
import { storiesOf } from '@storybook/react-native'

import { props as testDefaultProps } from './defaults'
import ExternalDependencies from './ExternalDependencies.component'

storiesOf('ExternalDependencies', module)
  .add('default', () => (
    <ExternalDependencies {...testDefaultProps} />
  ))
