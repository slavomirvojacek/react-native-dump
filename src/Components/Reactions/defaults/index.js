import { None } from '@threestup/monads'

import { issueDefault } from '../../Issue/defaults'

const props = {
  issue: issueDefault(),
  me: None
}

export {
  props
}
