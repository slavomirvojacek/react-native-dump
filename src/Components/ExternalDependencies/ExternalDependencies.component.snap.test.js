import React from 'react'

import ExternalDependencies from './ExternalDependencies.component'
import { props as testDefaultProps } from './defaults'

import { assertSnapshots } from '../../TestUtils/snapshot'

describe('ExternalDependencies Snapshots', () => {
  const config = [
    {
      props: {...testDefaultProps},
      desc: 'renders correctly'
    }
  ]

  assertSnapshots(ExternalDependencies, config)
})
