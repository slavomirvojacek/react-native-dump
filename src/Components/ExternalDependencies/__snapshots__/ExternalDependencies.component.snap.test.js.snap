// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalDependencies Snapshots renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
>
  Some Text
</Text>
`;
